iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1109,9 +1109,19 @@ public URI getURIForAuth(org.apache.hadoop.hive.metastore.api.Table hmsTable) th
         authURI.append(getPathForAuth(locationProperty.get()))
             .append(encodeString("/metadata/dummy.metadata.json"));
       } else {
-        Table table = IcebergTableUtil.getTable(conf, hmsTable);
-        authURI.append(getPathForAuth(((BaseTable) table).operations().current().metadataFileLocation(),
-            hmsTable.getSd().getLocation()));
+        try {
+          Table table = IcebergTableUtil.getTable(conf, hmsTable);
+          authURI.append(getPathForAuth(((BaseTable) table).operations().current().metadataFileLocation(),
+                  hmsTable.getSd().getLocation()));
+        } catch (NoSuchTableException ex) {
+          if (hmsTable.getSd() != null && !hmsTable.getSd().getLocation().isEmpty()) {
+            authURI.append(encodeString(URI.create(hmsTable.getSd().getLocation()).getPath()))
+                    .append(encodeString("/metadata/dummy.metadata.json"));
```

Comment:
nit: `"/metadata/dummy.metadata.json"` from L#1110 & L#1119 can be hardcoded in a final string variable.

Reply:
Sure. Done.